src/librustc/ty/subst.rs

```diff
@@ -184,6 +184,11 @@ pub type InternalSubsts<'tcx> = List<GenericArg<'tcx>>;
 pub type SubstsRef<'tcx> = &'tcx InternalSubsts<'tcx>;
 
 impl<'a, 'tcx> InternalSubsts<'tcx> {
+    pub fn closure(&'a self) -> ClosureSubsts<'a> {
```

Comment:
Final nit: Can we add a doc comment like so

---

Interpret these substitutions as the substitutions of a closure type. Closure substitutions have a particular structure controlled by the compiler that encodes information like the signature and closure kind; see `ClosureSubsts` struct for more comments.

Reply:
Maybe this should be called `as_closure()`? Seems strictly wordier, but maybe a bit more clear? I leave it to your judgement @csmoe =)